src/components/CategoricalColorPicker.tsx

```diff
@@ -0,0 +1,89 @@
+import React, { ReactElement, useMemo } from "react";
+import { Dataset } from "../colorizer";
+import { ColorPicker, Tooltip } from "antd";
+import { Color, ColorRepresentation } from "three";
+
+import { Color as AntColor } from "antd/es/color-picker/color";
+import styled from "styled-components";
+import { FlexRow, FlexRowAlignCenter } from "../styles/utils";
+
+type CategoricalColorPickerProps = {
+  dataset: Dataset | null;
+  featureName: string;
+  selectedPalette: Color[];
+  onChangePalette: (newPalette: Color[]) => void;
+  disabled?: boolean;
+};
+
+const defaultProps: Partial<CategoricalColorPickerProps> = {
+  disabled: false,
+};
+
+const ColorPickerContainer = styled(FlexRow)<{
+  $itemGap: string;
+  $itemWidth: string;
+  $maxItemsPerRow: string;
+}>`
+  gap: ${(props) => props.$itemGap};
+  flex-wrap: wrap;
+  margin-top: 8px;
+  margin-bottom: 8px;
+  margin-right: ${(props) => props.$itemGap};
+  max-width: calc(
+    ${(props) => `${props.$itemWidth} * ${props.$maxItemsPerRow} + ${props.$itemGap} * (${props.$maxItemsPerRow} - 1)`}
+  );
+
+  & > div {
+    // Container for both the color picker and the text label.
+    gap: 12px;
+    width: ${(props) => props.$itemWidth};
+    max-width: ${(props) => props.$itemWidth};
+    height: fit-content;
+  }
+
+  & > div > span {
+    // Text label, hide overflow as ellipsis
+    white-space: nowrap;
+    overflow: hidden;
+    text-overflow: ellipsis;
+  }
+`;
+
+export default function CategoricalColorPicker(inputProps: CategoricalColorPickerProps): ReactElement {
+  const props = { ...defaultProps, ...inputProps } as Required<CategoricalColorPickerProps>;
+  const categories = props.dataset?.getFeatureCategories(props.featureName);
```

Comment:
would there be any advantage to using categories as a prop instead of dataset+featurename?  The advantage I think of is that this component doesn't have to know anything about datasets but I suppose it doesn't matter too much.

Reply:
Hmm, I can only think of future benefits if we add more tooltip data or something. Changing this to `categories` makes sense to me.